clang/lib/Driver/ToolChains/AMDGPU.cpp

```diff
@@ -633,6 +633,17 @@ void amdgpu::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   else if (Args.hasArg(options::OPT_mcpu_EQ))
     CmdArgs.push_back(Args.MakeArgString(
         "-plugin-opt=mcpu=" + Args.getLastArgValue(options::OPT_mcpu_EQ)));
+
+  // If the user's toolchain has the 'include/amdgcn-amd-amdhsa/` path, we
+  // assume it supports the standard C libraries for the GPU and include them.
+  bool HasLibC = getToolChain().getStdlibIncludePath().has_value();
```

Comment:
maybe refactor as a member of ToolChain::addOffloadLibCArgs

Reply:
It's not really offload in this context, since it deals with `clang --target=amdgcn-amd-amdhsa`. But I could see putting it in `CommonArgs`.